AGENTS.md

```diff
@@ -0,0 +1,130 @@
+# Smithy-rs AI Agent Guide
+
+## Package Layout
+
+- **`codegen-core/`** - Shared codegen
+- **`codegen-server/`** - Server codegen
+- **`codegen-client/`** - Client codegen
+- **`rust-runtime/`** - Runtime libraries
+- **`codegen-server-test/`** - Server integration tests
+
+Protocol files: `codegen-{core,server}/.../protocols/`
+
+## preludeScope: Rust Prelude Types
```

Comment:
Probably clarify this is about codegen, perhaps nesting all of these under a common `Codegen` header.